From fbfc689850cd957247306b65e2370cc0145f639f Mon Sep 17 00:00:00 2001
From: Akshay Gupta <akshay.gupta@amd.com>
Date: Fri, 28 Feb 2025 05:13:21 +0000
Subject: [PATCH 5/6] sbrmi: Add revision 3.1 support for register size
 detection

Use the SBRMI revision to select the appropriate protocol for
CPUID and MCAMSR reads.
Derive the SBRMI register address size from the revision.

Reviewed-by: Muralidhara M K <muralidhara.mk@amd.com>
Signed-off-by: Akshay Gupta <akshay.gupta@amd.com>
Signed-off-by: Naveen Krishna Chatradhi <naveenkrishna.chatradhi@amd.com>
---
 drivers/apml/sbrmi-common.c | 2 ++
 drivers/apml/sbrmi.c        | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/apml/sbrmi-common.c b/drivers/apml/sbrmi-common.c
index 15c2aa345..f57ae2f6a 100644
--- a/drivers/apml/sbrmi-common.c
+++ b/drivers/apml/sbrmi-common.c
@@ -247,6 +247,7 @@ int rmi_mca_msr_read(struct apml_sbrmi_device *rmi_dev,
 
 		break;
 	case 0x21:
+	case 0x31:
 		ret = msr_datain_v21(rmi_dev, msg);
 		if (ret < 0)
 			goto exit_unlock;
@@ -359,6 +360,7 @@ int rmi_cpuid_read(struct apml_sbrmi_device *rmi_dev,
 			goto exit_unlock;
 		break;
 	case 0x21:
+	case 0x31:
 		ret = cpuid_datain_v21(rmi_dev, msg);
 		if (ret < 0)
 			goto exit_unlock;
diff --git a/drivers/apml/sbrmi.c b/drivers/apml/sbrmi.c
index 6e578f452..a34adddf1 100644
--- a/drivers/apml/sbrmi.c
+++ b/drivers/apml/sbrmi.c
@@ -391,6 +391,7 @@ static int sbrmi_i2c_identify_reg_addr_size(struct i2c_client *i2c, u32 *size, u
 		*size = SBRMI_REG_ADDR_SIZE_DEF;
 		break;
 	case 0x21:
+	case 0x31:
 		*size = SBRMI_REG_ADDR_SIZE_TWO_BYTE;
 		break;
 	default:
@@ -497,6 +498,7 @@ static int sbrmi_i3c_identify_reg_addr_size(struct i3c_device *i3cdev, u32 *size
 		*size = SBRMI_REG_ADDR_SIZE_DEF;
 		break;
 	case 0x21:
+	case 0x31:
 		*size = SBRMI_REG_ADDR_SIZE_TWO_BYTE;
 		break;
 	default:
-- 
2.43.0

